packages/ra-core/src/ui/delete/Delete.js

```diff
@@ -174,72 +120,25 @@ export class Delete extends Component {
         const defaultTitle = translate('ra.page.delete', {
             name: `${resourceName}`,
             id,
-            data,
+            record,
         });
-        const titleElement = data ? (
-            <Title title={title} record={data} defaultTitle={defaultTitle} />
-        ) : (
-            ''
-        );
 
-        return (
-            <div className={className} {...rest}>
-                <Card style={{ opacity: isLoading ? 0.8 : 1 }}>
-                    <Header
-                        title={titleElement}
-                        actions={actions}
-                        actionProps={{
-                            basePath,
-                            data,
-                            hasEdit,
-                            hasList,
-                            hasShow,
-                        }}
-                    />
-                    <form onSubmit={this.handleSubmit}>
-                        <CardContent>
-                            <Typography>
-                                {translate('ra.message.are_you_sure')}
-                            </Typography>
-                        </CardContent>
-                        <Toolbar disableGutters={true}>
-                            <Button
-                                variant="raised"
-                                type="submit"
-                                color="primary"
-                                className={classes.button}
-                            >
-                                <ActionCheck
-                                    className={classes.iconPaddingStyle}
-                                />
-                                {translate('ra.action.delete')}
-                            </Button>
-                            &nbsp;
-                            <Button
-                                variant="raised"
-                                onClick={this.goBack}
-                                className={classes.button}
-                            >
-                                <AlertError
-                                    className={classes.iconPaddingStyle}
-                                />
-                                {translate('ra.action.cancel')}
-                            </Button>
-                        </Toolbar>
-                    </form>
-                </Card>
-            </div>
-        );
+        return children({
+            basePath,
+            defaultTitle,
+            handleSubmit: this.handleSubmit,
+            isLoading,
+            record,
+            goBack: this.goBack,
+        });
```

Comment:
so you're no longer passing rest props to children, not good

Reply:
The `ra-ui-materialui` `Delete` component does